File: angle-core/modules/matrixsolve.go
package modules

import (
	"encoding/json"
	"os/exec"
)

// RunMatrixsolve 调用 Python 实现矩阵求解 || Call Python for matrix solving
func RunMatrixsolve(input map[string]interface{}) map[string]interface{} {
	// 参数转为 JSON || Marshal params to JSON
	inBytes, _ := json.Marshal(input)
	cmd := exec.Command("../env_image/bin/python3.10", "../env_image/py_modules/matrixsolve.py")
	stdin, _ := cmd.StdinPipe()
	stdout, _ := cmd.StdoutPipe()
	cmd.Start()
	stdin.Write(inBytes)
	stdin.Close()
	outBytes := make([]byte, 4096)
	n, _ := stdout.Read(outBytes)
	cmd.Wait()
	var result map[string]interface{}
	json.Unmarshal(outBytes[:n], &result)
	return result
}

File: env_image/py_modules/matrixsolve.py
import sys, json
import numpy as np

# 读取输入参数 || Read input params
params = json.load(sys.stdin)
A = np.array(params.get('A', [[1,0],[0,1]]))
b = np.array(params.get('b', [0,0]))

try:
    # 求解线性方程组 Ax = b || Solve Ax = b
    x = np.linalg.solve(A, b)
    result = {'result': x.tolist()}
except Exception as e:
    result = {'error': str(e)}

# 输出结果为 JSON || Output result as JSON
json.dump(result, sys.stdout) 